Portable clipboard handling

chocolateboy opened the pull request:
This PR closes #143 by providing sensible defaults for the `clipboard_copy_cmd` option on:

- macOS (using pbcopy)
- Windows (using clip)
- Linux etc. (using xclip or xsel)

There are issues (chiefly https://github.com/asweigart/pyperclip/issues/67, which causes a segfault on Python 3) which make pyperclip unusable on Linux in its current form, so a small helper class is used instead.

chocolateboy committed:
portable clipboard handling (closes #143)

provide sensible defaults for the clipboard_copy_cmd option on:

    - macOS (using pbcopy)
    - Windows (using clip)
    - Linux etc. (using xclip or xsel)

saulpw commented:
Thanks for doing this, @chocolateboy!  Sorry for the delay in merging, we're at PyCon this week and haven't had time to test the change ourselves.  We'll merge it as soon as we can and then it will be in the next release.  Thanks again!

chocolateboy committed:
restrict checks for platform-specific clipboard-commands to their platforms

only check for clip on Windows and pbcopy on macOS.
this prevents e.g. ~/bin/clip being mistaken for the
Windows command on Linux systems

anjakefala commented on visidata/clipboard.py:
```diff
@@ -20,35 +24,76 @@
+commands = {
+    ('clip', 'win32'):    [],                                      # Windows Vista+
+    ('pbcopy', 'darwin'): ['w'],                                   # macOS
+    ('xclip', None):      ['-selection', 'clipboard', '-filter'],  # Linux etc.
```
Hey @chocolateboy! I was wondering about some of the changes you made here.

From what I can tell, within`xclip` the `-filter` flag seems to make some output from xclip overlay the VisiData interface. So i have a preference for removing it.

Additionally, I was wondering about the choice for `primary` -> `clipboard ` for both `xclip` and `xsel`? They seem to work well but what is the underlying difference between them?

Otherwise, works beautifully and I am really glad you did the work for this.. =)

chocolateboy replied:
If `-filter` is omitted, xclip runs as a daemon, which [blocks when run in a pipeline](https://github.com/asweigart/pyperclip/issues/116#issuecomment-387825778). This may not matter in VisiData (I haven't run it in a pipeline), but it shouldn't hurt. I think a better solution than launching xclip as a daemon every time would be to ensure the command's stdout is consumed and discarded (which is what I assumed was happening).

`clipboard` is the system clipboard targeted by <kbd>Ctrl-C</kbd>, <kbd>Ctrl-V</kbd>, and <kbd>Ctrl-X</kbd>. This is the standard clipboard used by browsers, text editors etc i.e. the apps most likely to be on the receiving end of pastes from VisiData. `primary` is the X clipboard accessible by clicking the left and middle mouse buttons. I believe that use case is less common (my guess is that most users are unaware of it), particularly for a keyboard-oriented tool, but anyone who has a preference for it can still, of course, configure it via the `clipboard_copy_cmd` option.

chocolateboy replied:
Just pushed a fix which discards the clipboard command's stdout.

anjakefala replied:
Thanks for elaborating @chocolateboy and for the fix which discards the clipboard's stdout! Looks good.

chocolateboy committed:
discard the clipboard command's stdout

this prevents it being displayed in the UI
